---
course: NGS for evolutionary biologists: from basic scripting to variant calling
title: Tutorial - Variant calling SNPs
requires:
author:  Chiara Batini, Pille Hallast  
time:
---
------------
> #### Learning Objectives
------------


#Variant Calling
#Practical Handbook

*Summary*
During this practical you will  
- identify variants
- filter variants

## Data Files

This practical will continue from the day3_mapping_BAM_refinement practical. We will use the final bam file created yesterday to perform the variant calling.  

- bam file: `library_final_sorted.bam`
- reference genome file: `Saccharomyces_cerevisiae.EF4.68.dna.toplevel.fa`

**Are the reference index and dictionary in the same directory as the reference file?**


If you are starting your analyses directly from a `.bam` file created by someone else, make sure you have the same reference genome they have used for the alignment. It is essential that the contigs in the reference are the same, in number, length and ID, to those used in the `.bam` file.

## Software Used:

- [Samtools / bcftools](http://samtools.sourceforge.net/samtools.shtml) are collections of utilities for manipulating sam / vcf files respectively.  

- [Genome Analysis Toolkit (GATK)](http://www.broadinstitute.org/gatk/ ) software is designed for variant discovery and genotyping.

- [VCFtools](http://vcftools.sourceforge.net/) is a set of scripts to manipulate vcf files.

- [vcflib](https://github.com/ekg/vcflib) is another set of scripts to manipulate vcf files.

